protostar/starknet/cairo1_parser.py

```diff
@@ -0,0 +1,60 @@
+# pylint: disable="protected-access"
+from dataclasses import dataclass
+from pathlib import Path
+from typing import Any
+import json
+
+from starkware.cairo.lang.compiler.program import Program
+from starkware.starkware_utils.marshmallow_dataclass_fields import IntAsHex
+
+
+@dataclass(frozen=True)
+class CairoHintCode:
+    code: str
+    accessible_scopes: list[Any]
+    flow_tracking_data: Any
+
+
+@dataclass(frozen=True)
+class TestCase:
+    name: str
+    offset: int
+
+
+@dataclass(frozen=True)
+class TestSuite:
+    path: Path
+    test_cases: list[TestCase]
+    program: Program
+
+
+def parse_test_suite(path: Path, json_raw: Any) -> TestSuite:
+    json_dict = json.loads(json_raw)
```

Comment:
Maybe `jsonschema` should be used instead. The schema could be used as a form of documentation.

Reply:
I think it is too much work, format is defined by cairo compiler. I would add a link to relevant code but i am scared that it will quickly become obsolete

Reply:
I will add source code in docstring, I think It will be pretty self-explaining